elasticsearch-stress-test.py

```diff
@@ -316,60 +298,68 @@ def print_stats():
     print("")
 
 
-def print_stats_worker():
-
+def print_stats_worker(STARTED_TIMESTAMP):
     # Create a conditional lock to be used instead of sleep (prevent dead locks)
     lock = Condition()
 
     # Acquire it
     lock.acquire()
 
     # Print the stats every STATS_FREQUENCY seconds
-    while not has_timeout():
+    while (not has_timeout(STARTED_TIMESTAMP)) and (not shutdown_event.is_set()):
 
         # Wait for timeout
         lock.wait(STATS_FREQUENCY)
 
         # To avoid double printing
-        if not has_timeout():
-
+        if not has_timeout(STARTED_TIMESTAMP):
             # Print stats
-            print_stats()
+            print_stats(STARTED_TIMESTAMP)
 
 
 def main():
-    # Define the globals
-    global documents_templates
-    global indices
-    global STARTED_TIMESTAMP
-    global es
+    clients = []
+    all_indecies = []
 
-    try:
-        # Initiate the elasticsearch session
-        es = Elasticsearch(ES_ADDRESS)
+    # Set the timestamp
+    STARTED_TIMESTAMP = int(time.time())
 
-    except:
-        print("Could not connect to elasticsearch!")
-        sys.exit(1)
+    for esaddress in args.es_address:
+        print("")
+        print(esaddress)
+        try:
+            # Initiate the elasticsearch session
+            es = Elasticsearch(esaddress)
 
-    print("Generating documents and workers.. "),
+        except Exception as e:
+            print("Could not connect to elasticsearch!")
+            sys.exit(1)
 
-    # Generate the clients
-    clients = generate_clients()
+        # Generate docs
+        documents_templates = generate_documents()
+        fill_documents(documents_templates)
 
-    # Generate docs
-    documents_templates = generate_documents()
-    fill_documents()
+        print("Done!")
+        print("Creating indices.. ")
 
-    print("Done!")
-    print("Creating indices.. "),
+        indices = generate_indices(es)
+        all_indecies.extend(indices)
 
-    indices = generate_indices()
+        try:
+            #wait for cluster to be green
+            es.cluster.health(wait_for_status='green', master_timeout='600s', timeout='600s')
+        except Exception as e:
+            print("Cluster timeout....")
+            print("Cleaning up created indices.. "),
 
-    print("Done!")
+            cleanup_indices(es, indices)
+            continue
+
+        print("Generating documents and workers.. ")  # Generate the clients
+        clients.extend(generate_clients(es, indices, STARTED_TIMESTAMP))
```

Comment:
there is a try catch in the function cleanup_indices. It will continue if the index isn't created or the cluster doesn't answer